### DISCUSSION TITLE
What is the purpose of the first part of the CSRF token in randomize()/derandomize() methods?

### DISCUSSION THREAD

**Discussion by denis-gol** [symfony/symfony#49127] [Q&A] [answered]
https://github.com/symfony/security-csrf/blob/6.2/CsrfTokenManager.php , lines: 117...133

it is created with code: "substr(md5($key), 0, 1 + (\ord($key[0]) % 32))"
but no verifying in derandomize():120

During login I can change first part of csrf-token and Symfony will not check in anyway. Login will be succeed.

**Comment by stof** [comment, marked as answer]
https://github.com/symfony/security-csrf/commit/4c6906a370bc9be11bad5d86dc148fc4fa71c307

**Reply by stof** [reply]
The original PR is https://github.com/symfony/symfony/pull/39919
